bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/scanner/RowMergerTest.java

```diff
@@ -72,135 +66,132 @@ public void resultsAreReadable() throws IOException {
     }
   }
 
-  @Test
-  public void multipleChunksAreMerged() throws IOException {
-    String rowKey = "row-1";
-
-    ReadRowsResponse response = createReadRowsResponse(rowKey, Family1_c1_CHUNK);
-    ReadRowsResponse response2 = createReadRowsResponse(rowKey, Family1_c2_CHUNK);
-    ReadRowsResponse response3 = createReadRowsResponse(rowKey, Family2_null_CHUNK);
-    ReadRowsResponse response4 = createReadRowsResponse(rowKey, COMPLETE_CHUNK);
-
-    Iterator<ReadRowsResponse> iterator = getIterator(response, response2, response3, response4);
-
-    Row resultRow = RowMerger.readNextRow(iterator);
-    Assert.assertEquals(2, resultRow.getFamiliesCount());
+  private Iterator<ReadRowsResponse> getIterator(ReadRowsResponse ... responses) {
+    return Arrays.asList(responses).iterator();
+  }
 
-    Map<String, Family> familyMap = new HashMap<>();
-    for (Family family : resultRow.getFamiliesList()) {
-      familyMap.put(family.getName(), family);
+  private void matchResponses(ReadRowsResponse[] responses, RowMatcher[] expectedRows) {
+    Iterator<ReadRowsResponse> iterator = getIterator(responses);
+    for (int i = 0; i < expectedRows.length; ++i) {
+      Assert.assertTrue("More expected rows than supplied responses", iterator.hasNext());
+      Row result = RowMerger.readNextRow(iterator);
+      Assert.assertThat("Expected row does not match!", result, expectedRows[i]);
```

Comment:
Yes; you get an AssertionError like:

java.lang.AssertionError: Expected row does not match!
Expected: row with key matching <<ByteString@379619aa size=5>> and column families: family name: "Family3", family name: "Family1""column qualifier "<<ByteString@cac736f size=2>> with <1> cells matching: an instance of com.google.bigtable.v1.Cell, "column qualifier "<<ByteString@5e265ba4 size=2>> with <1> cells matching: an instance of com.google.bigtable.v1.Cell
     but: was <key: "row-1"
families {
  name: "Family2"
}
families {
  name: "Family1"
  columns {
    qualifier: "c1"
    cells {
      timestamp_micros: 100
      value: "\2324.\223)dF\241\344\312"
    }
  }
  columns {
    qualifier: "c2"
    cells {
      timestamp_micros: 100
      value: "\210\302w\251Tr\271\031>\276+\315\362\344V$\247\000\322\305\275\364pc6\250\026)\276 \232\333%e\233\302\356\270A\206T\376S\226\260\264#q\317h\265\016\370j\244\363L\312\335\002\273r\034\214\'l\342j\2336\230\022\2433\362\017\372Dl\306wu\267Q\310\255\306D\356=\'}\322\341\364p(<\243\213"
    }
  }
}
>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at com.google.cloud.bigtable.grpc.scanner.RowMergerTest.matchResponses(RowMergerTest.java:78)
	at com.google.cloud.bigtable.grpc.scanner.RowMergerTest.multipleChunksAreMerged(RowMergerTest.java:85)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[etc.]


It is not the greatest error in the world. (Especially don't care for the ByteString representation.) But making that better seems of limited value at the moment...